v21.2/local-testing.md

```diff
@@ -8,16 +8,37 @@ This page documents best practices for unit testing applications built on Cockro
+|--------------------------------------------------------------+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `kv.raft_log.disable_synchronization_unsafe`                 | `true`    | Improves performance by not syncing data to disk. Data is lost if a node crashes.                                                                                                    |
+| `kv.range_merge.queue_interval`                              | `50ms`    | Frequent CREATE TABLE or DROP TABLE creates extra ranges, which we want to merge more quickly. In real usage, range merges are rate limited because they require rebalancing.        |
+| `jobs.registry.interval.gc`                                  | `30s`     | CockroachDB executes internal queries that scan the jobs table. More schema changes create more jobs, which we can delete faster to make internal job queries faster.                |
+| `jobs.registry.interval.cancel`                              | `180s`    | Timing of an internal task that queries the jobs table. For testing, the default is too fast.                                                                                        |
+| `jobs.retention_time`                                        | `15s`     | More schema changes create more jobs, which affects job query performance. We don’t need to retain jobs during testing and can set a more aggressive delete policy.                  |
+| `schemachanger.backfiller.buffer_increment`                  | `128 KiB` | During table backfills, we fill up buffers which have a large default. A lower setting reduces memory usage.                                                                         |
+| `sql.catalog.unsafe_skip_system_config_trigger`              | `true`    | Turn off gossiping the system config span, which is not necessary in a test environment. (Available in v22.1 and later.)                                                             |
+| `sql.stats.automatic_collection.enabled`                     | `false`   | Turn off statistics collection, since automatic statistics contribute to table contention alongside schema changes. Each schema change triggers an asynchronous auto statistics job. |
+| `ALTER RANGE default CONFIGURE ZONE USING "gc.ttlseconds"`   | `5s`      | Faster descriptor cleanup. For more information, see [`ALTER RANGE`](alter-range.html).                                                                                              |
+| `ALTER DATABASE system CONFIGURE ZONE USING "gc.ttlseconds"` | `5s`      | Faster jobs table cleanup. For more information, see [`ALTER DATABASE`](alter-database.html).                                                                                        |
```

Comment:
since added `kv.range_split.by_load_merge_delay`  in v21.2 ? or upcoming in v22.1 ?

Added the SQL statement with everything else (pending your answer re: `kv.range_split.by_load_merge_delay`)

Reply:
(nice, this flushed out a little doc bug too - I added the CONFIGURE ZONE statements to the list as well and they take `5` not `5s` as the arg)